File: ios/AddCardRequestHandler.m
//
//  AddCardRequestHandler.m
//  WhalesApp
//
//  Created by VZ on 17/7/24.
//

#import "AddCardRequestHandler.h"

@implementation AddCardRequestHandler

- (instancetype)initWithResolver:(RCTPromiseResolveBlock)resolve
                        rejecter:(RCTPromiseRejectBlock)reject
                          cardId:(NSString *)cardId
                           token:(NSString *)token
                         network:(NSString *)network {
  self = [super init];
  if (self) {
    _cardId = cardId;
    _token = token;
    _resolver = resolve;
    _rejecter = reject;
    _network = network;
  }
  return self;
}

@end

File: ios/AddCardRequestHandler.swift
//
//  AddCardRequestHandler.swift
//  WhalesApp
//
//  Created by VZ on 1/8/24.
//

import Foundation
import React

@objc(AddCardRequestHandler)
class AddCardRequestHandler: NSObject {
    var cardId: String
    var token: String
    var resolver: RCTPromiseResolveBlock
    var rejecter: RCTPromiseRejectBlock
    var isTestnet: Bool

    @objc
    init(resolver: @escaping RCTPromiseResolveBlock, rejecter: @escaping RCTPromiseRejectBlock, cardId: String, token: String, isTestnet: Bool) {
        self.cardId = cardId
        self.token = token
        self.resolver = resolver
        self.rejecter = rejecter
        self.isTestnet = isTestnet
    }
}
